front/hooks/useFileUploaderService.ts

```diff
@@ -0,0 +1,203 @@
+import type { SupportedContentFragmentType } from "@dust-tt/types";
+import { isSupportedTextContentFragmentType } from "@dust-tt/types";
+import { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { extractTextFromPDF } from "@app/lib/client/handle_file_upload";
+import { getMimeTypeFromFile } from "@app/lib/file";
+
+interface FileContent {
```

Comment:
**Question:** 
Kindda feels like we are starting to have a lot of different types to handle Files/ContentFragment makes it hard to understand. WDYT? Dunno if we could have `UploadedContentFragment` or `FileContent`?

Reply:
The goal of this initiative is too split file upload from content fragment.